datasetinsights/storage/checkpoint.py

```diff
@@ -162,32 +169,47 @@ def save(self, estimator, epoch=None):
 
         return full_cloud_path
 
-    def load(self, estimator, full_cloud_path):
-        """ Load estimator from checkpoint files on GCS
 
-        Args:
-            estimator (datasetinsights.estimators.Estimator):
-                datasetinsights estimator object.
-            path: full path to the checkpoint file
-        """
-        bucket, object_key = gcs_bucket_and_path(full_cloud_path)
-        filename = os.path.basename(object_key)
-        path = os.path.join(self._tempdir, filename)
-        logger.debug(f"Downloading estimator from {full_cloud_path} to {path}")
-        self._client.download(bucket, object_key, path)
+def load_local(estimator, path):
+    """ loads estimator checkpoints from a local path """
+    estimator.load(path)
 
-        self._checkpointer.load(estimator, path)
+    return path
 
 
-class DistributedEstimatorCheckpoint:
-    def __init__(self, is_master, estimator_checkpoint):
-        self.is_master = is_master
-        self.estimator_checkpoint = estimator_checkpoint
+def load_from_gcs(estimator, full_cloud_path):
+    """ Load estimator from checkpoint files on GCS
 
-    def save(self, estimator, epoch=None):
-        if self.is_master:
-            self.estimator_checkpoint.save(estimator, epoch)
+    Args:
+        estimator (datasetinsights.estimators.Estimator):
+            datasetinsights estimator object.
+        full_cloud_path: full path to the checkpoint file
+    """
+    bucket, object_key = gcs_bucket_and_path(full_cloud_path)
+    filename = os.path.basename(object_key)
+    temp_dir = tempfile.TemporaryDirectory().name
```

Comment:
How do you make sure to cleanup `TemporaryDirectory` after model artifact is loaded into memory? Using context manager might help. See these [examples](https://docs.python.org/3/library/tempfile.html#tempfile-examples).

Reply:
Since we are returning the path here, do we not want to keep the directory instead of deleting it?